src/Download/DownloadMenu.tsx

```diff
@@ -171,61 +146,77 @@ const DownloadMenu = ({
 		].filter(da => da.include);
 	};
 
-	React.useEffect(() => {
+	useEffect(() => {
 		if (showZipModal && currentBct && !zipCancelled.current) {
 			setZipTotal(plants.length);
 			setZipCurrent(0);
 			const cb = async () => {
 				try {
-					const plantBlobPairs = await Promise.all(
-						plants.map(async (p: Plant) => {
-							let b: any;
-							try {
-								if (zipCancelled.current) {
-									throw new Error("Download cancelled");
-								}
-								b = await pdf(
-									<BenchCardDocument
-										benchCardTemplate={currentBct}
-										plant={p}
-										region={searchCriteria.city.region}
-										waterUseByCode={data.waterUseByCode}
-									/>
-								).toBlob();
-								let current = 0;
-								setZipCurrent((i) => {
-									current = i === 0 ? 1 : i + 1;
-									return current;
-								});
-								console.log(
-									`generated bench card ${current} of ${plants.length}`
-								);
-							} catch (e: any) {
-								if (e.message === "Download cancelled") {
-									console.log(e.message);
-								} else {
-									console.error(e);
-								}
+					const plantBlobPairs: [Plant, any][] = [];
+
+					// Process plants sequentially instead of in parallel
```

Comment:
why? was the old version of parallel processing causing problems?

Reply:
Yeah, the modal had a progress bar that wasn't getting updated because state updates were clobbering each other.